### PR TITLE
[Test refactor 5/5] Build docker images

### PR DESCRIPTION
**Pull request by LysandreJik**
These docker images will be built on a daily basis to be re-used by the CI.

Additional docker images will be built for Torch versions in [1.4-1.10] and TensorFlow versions in [2.3-2.7].

### CODE REVIEW

File: docker/transformers-all-latest-gpu/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+RUN apt update
+RUN apt install -y git libsndfile1-dev tesseract-ocr espeak-ng python3 python3-pip ffmpeg
+RUN python3 -m pip install --no-cache-dir --upgrade pip
+
+ARG REF=master
+RUN git clone https://github.com/huggingface/transformers && cd transformers && git checkout $REF
+RUN python3 -m pip install --no-cache-dir -e ./transformers[dev,onnxruntime]
+
+RUN python3 -m pip install --no-cache-dir -U torch tensorflow
+RUN python3 -m pip uninstall -y flax jax
```
Comment (patrickvonplaten) [inline docker/transformers-all-latest-gpu/Dockerfile:15]: flax  jax is not included yet no? => sounds good to me! 

Don't we need https://github.com/huggingface/transformers/blob/86119c115496ca773b82f8a1c8cbf2d4e44149fa/.github/workflows/self-scheduled.yml#L39 here as well though?

File: docker/transformers-pytorch-gpu/Dockerfile [hunk 2]
```diff
@@ -1,30 +1,25 @@
-FROM nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04
+FROM nvidia/cuda:11.2.2-cudnn8-runtime-ubuntu20.04
```
Comment (patrickvonplaten) [inline docker/transformers-pytorch-gpu/Dockerfile:1]: Think we need https://github.com/huggingface/transformers/blob/86119c115496ca773b82f8a1c8cbf2d4e44149fa/.github/workflows/self-scheduled.yml#L39 here as well no?

**Review by patrickvonplaten — APPROVED** [review]
Awesome! Happy to merge as it is and then to quickly adapt if errors pop up

File: docker/transformers-all-latest-gpu/Dockerfile (hunk 1 above)
Comment (sgugger) [reply docker/transformers-all-latest-gpu/Dockerfile:15]: It's two lines below Patrick ;-)

File: docker/transformers-pytorch-gpu/Dockerfile (hunk 2 above)
Comment (sgugger) [reply docker/transformers-pytorch-gpu/Dockerfile:1]: Same, it's on line 21 ;-)

**Review by sgugger — APPROVED** [review]
Great work!

**Review by sgugger — APPROVED** [review]

File: docker/transformers-all-latest-gpu/Dockerfile (hunk 1 above)
Comment (patrickvonplaten) [reply docker/transformers-all-latest-gpu/Dockerfile:15]: indeed thanks!

**Pull request merged**